group-size-refinement-list/react-instantsearch/src/lib/GroupSizeRefinementList.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from "react";
+import { Highlight, connectRefinementList } from "react-instantsearch-dom";
+
+import "./GroupSizeRefinementList.css";
+
+class GroupSizeRefinementList extends Component {
+    constructor(props) {
+        super(props);
+        const showMore = this.props.translations && this.props.translations.showMore ?
+            this.props.translations.showMore :
+            (expanded) => expanded ? 'Show less' : 'Show more';
+
+        this.state = {
+            showMore,
+            sortGroupByNbResults: props.sortGroupByNbResults !== undefined ? props.sortGroupByNbResults : true,
+            sortSizesByNbResults: props.sortSizesByNbResults !== undefined ? props.sortSizesByNbResults : true,
+            expanded: !props.showMore,
+            nbGroups: props.nbGroups ? props.nbGroups : 1
+        };
+    }
+
+    render() {
+        const { expanded, showMore } = this.state;
+
+        let inclusionArrays = [];
+        const patterns = [...this.props.patterns, /.*/im]
+        let nbGroups = patterns.length
+        let sizeGroups = patterns.reduce((acc, regex) => {
+            const sizeGroup = this.props.items.reduce((prevHit, hit) => {
+                if (regex.test(hit.label.split(';')[0]) && !inclusionArrays.includes(hit)) {
+                    return { count: prevHit.count + hit.count, hits: [...prevHit.hits, hit] }
+                }
+                return prevHit
+            }, { count: 0, hits: [] });
+            if (sizeGroup.count > 0) {
+                inclusionArrays.push(...sizeGroup.hits)
+                return [...acc, sizeGroup]
+            }
+            return acc
+        }, []);
+
+        //Display the group that has the biggest count on top or number of choices if tie
+        if (this.state.sortGroupByNbResults)
```

Comment:
You can put this in a separate function so that this becomes one line
`const sortedSizeGroups = sortSizeGroupsByCount(sizeGroups)`
And you wouldn't even need to add a comment to explain what it's doing.